app/gui2/src/components/GraphEditor/widgets/WidgetPort.vue

```diff
@@ -36,13 +36,15 @@ const navigator = injectGraphNavigator()
 const tree = injectWidgetTree()
 const selection = injectGraphSelection(true)
 
-const isHovered = computed(() => selection?.hoveredPort === props.input.portId)
-
 const hasConnection = computed(() => graph.isConnectedTarget(portId.value))
 const isCurrentEdgeHoverTarget = computed(
-  () => isHovered.value && graph.unconnectedEdge != null && selection?.hoveredPort === portId.value,
+  () =>
+    graph.unconnectedEdge?.anchor.type === 'mouse' &&
```

Comment:
Perhaps instead of `unconnectedEdge` we could have two fields: `editedEdge` and `componentBrowserEdge` of more specific type - here, for example, you would not have to check anchor type.